src/infrastructure/tracer.ts

```diff
@@ -0,0 +1,15 @@
+import ddTrace from 'dd-trace';
+
+const tracer = ddTrace.init({
+	service: process.env.FIGMA_SERVICE_NAME || '',
+	profiling: true,
```

Comment:
**question:** @elainefigma , I apologise for not verifying the changes on our environment -- I will do this early tomorrow. Please, could you help me to understand what makes `dd-tracer` to switch between the "no-op" behavior and integration with Datadog?

From what I see in the [docs](https://docs.datadoghq.com/tracing/trace_collection/library_config/nodejs/#instrumentation), Datadog enables many features by default. Sorry if I am missing something: I haven't set up Datadog before.

<img width="830" alt="image" src="https://github.com/atlassian-labs/figma-for-jira/assets/141381034/95dc3f04-b053-4f58-930e-571689b73e3f">

Reply:
Apologies for the confusion - it's not entirely a no-op. [Here](https://docs.datadoghq.com/tracing/trace_collection/?tab=containers#install-and-configure-the-agent) is a diagram for reference. 

To clarify, in our setup, when we deploy figma-for-jira, we also spin up a Datadog agent. When you add the Datadog tracing library to your code, it instruments the service and sends traces to the Datadog Agent. The Agent then sends the traces to the Datadog backend to be displayed in the UI.

In the dev setup and the Jira setup, there is no Datadog agent. In these environments, we send traces, and nothing happens since the Datadog agent is not set up.

Reply:
Got it! Thank you very much for the clarification, @elainefigma -- it all makes sense now!

This being the case, I also recommend disabling `dd-trace` features completely for non-Figma environments. We could achieve this by using the [DD_TRACE_ENABLED](https://docs.datadoghq.com/tracing/trace_collection/library_config/nodejs/#instrumentation) environment variable.

Please, consider adding `DD_TRACE_ENABLED=false` to our `.env.example`:

```
# Optional. These env variables are used to configure Datadog for tracing.
DD_TRACE_ENABLED=false
DD_SERVICE=figma-for-jira
DD_ENV=dev
```

Reply:
Just for sharing... It seems that Datadog uses this env variable to choose between the "noop" and proxy-based tracing implementations:

https://github.com/DataDog/dd-trace-js/blob/0999ce6ab2bcde6ab77ad502f03a7c203667050d/packages/dd-trace/src/index.js#L8

Reply:
That suggestion makes sense to me - thank you for looking into this!